### PR TITLE
Block oracle improvements

### PR DESCRIPTION
**Pull request by noamnelke**
There were a couple of comments in #778 that this fixes.

### CODE REVIEW

File: mesh/activation.go [hunk 1]
```diff
@@ -9,16 +9,13 @@ import (
 	"github.com/spacemeshos/go-spacemesh/nipst"
 )
 
-//todo: choose which type is VRF
-type Vrf string
-
 type NodeId struct {
```
Comment (zalmen) [inline mesh/activation.go:12]: There should be one definition of `NodeId` and it should be higher in the file hierarchy, not here.

**Review by zalmen — DISMISSED** [review]

File: mesh/activation.go (hunk 1 above)
Comment (zalmen) [reply mesh/activation.go:12]: The Hare eligibility will also need to use it, no reason for it to import this package

Comment (noamnelke) [reply mesh/activation.go:12]: Created https://github.com/spacemeshos/go-spacemesh/issues/790

File: oracle/blockoracle_test.go [hunk 2]
```diff
@@ -10,9 +10,14 @@ import (
 )
 
 var atxID = mesh.AtxId{Hash: [32]byte{1, 3, 3, 7}}
-var nodeID = mesh.NodeId{
-	Key: "key",
-	Vrf: "vrf",
+var nodeID, vrfSigner = generateNodeIDAndSigner()
+
+func generateNodeIDAndSigner() (mesh.NodeId, *crypto.VRFSigner) {
+	publicKey, privateKey, _ := crypto.GenerateVRFKeys()
```
Comment (antonlerner) [inline oracle/blockoracle_test.go:16]: comment that for each run a different key will be created at the moment

**Review by antonlerner — APPROVED** [review]

**Pull request merged**